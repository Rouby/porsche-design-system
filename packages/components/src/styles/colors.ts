import type { Theme } from '@porsche-design-system/utilities-v2';
import { isThemeDark } from '../utils';

type ThemedColor =
  | 'primaryColor'
  | 'primaryColorDarken'
  | 'backgroundColor'
  | 'backgroundColorDarken'
  | 'backgroundColorLighten'
  | 'backgroundSurfaceColor'
  | 'backgroundSurfaceColorDarken'
  | 'backgroundSurfaceColorLighten'
  | 'contrastLowColor'
  | 'contrastMediumColor'
  | 'contrastHighColor'
  | 'contrastHighColorDarken'
  | 'contrastHighColorLighten'
  | 'hoverColor'
  | 'hoverColorDarken'
  | 'activeColor'
  | 'focusColor'
  | 'disabledColor'
  | 'errorColor'
  | 'errorColorDarken'
  | 'errorSoftColor'
  | 'errorSoftColorDarken'
  | 'errorSoftColorLighten'
  | 'successColor'
  | 'successColorDarken'
  | 'successSoftColor'
  | 'successSoftColorDarken'
  | 'successSoftColorLighten'
  | 'warningColor'
  | 'warningSoftColor'
  | 'warningSoftColorDarken'
  | 'warningSoftColorLighten'
  | 'infoColor'
  | 'infoSoftColor'
  | 'infoSoftColorDarken'
  | 'infoSoftColorLighten';

export type ThemedColors = { [key in ThemedColor]: string };

/* Auto Generated Start */
const themeLight = {
  primaryColor: '#010205',
  primaryColorDarken: '#000000',
  backgroundColor: '#FFF',
  backgroundColorDarken: '#E0E0E0',
  backgroundColorLighten: '#FFFFFF',
  backgroundSurfaceColor: '#EEEFF2',
  backgroundSurfaceColorDarken: '#CBCED7',
  backgroundSurfaceColorLighten: '#FFFFFF',
  contrastLowColor: '#D8D8DB',
  contrastMediumColor: '#949598',
  contrastHighColor: '#535457',
  contrastHighColorDarken: '#353638',
  contrastHighColorLighten: '#717276',
  hoverColor: 'rgba(148, 149, 152, .18)',
  hoverColorDarken: '#75767A',
  activeColor: 'rgba(148, 149, 152, 0.20)',
  focusColor: '#1A44EA',
  disabledColor: '#949598',
  errorColor: '#E7323B',
  errorColorDarken: '#C51720',
  errorSoftColor: '#FCE8E9',
  errorSoftColorDarken: '#EED7D9',
  errorSoftColorLighten: '#FFFFFF',
  successColor: '#32B85B',
  successColorDarken: '#258843',
  successSoftColor: '#EBFAF0',
  successSoftColorDarken: '#DDEAE1',
  successSoftColorLighten: '#FFFFFF',
  warningColor: '#FECC1B',
  warningSoftColor: '#FFF9E6',
  warningSoftColorDarken: '#F5ECD2',
  warningSoftColorLighten: '#FFFFFF',
  infoColor: '#1E5BEB',
  infoSoftColor: '#E8EEFD',
  infoSoftColorDarken: '#D6DEF0',
  infoSoftColorLighten: '#FFFFFF'
};

const themeDark = {
  primaryColor: '#FBFCFF',
  primaryColorDarken: '#BECEFF',
  backgroundColor: '#0E0E12',
  backgroundColorDarken: '#000000',
  backgroundColorLighten: '#292934',
  backgroundSurfaceColor: '#212225',
  backgroundSurfaceColorDarken: '#040405',
  backgroundSurfaceColorLighten: '#3E4045',
  contrastLowColor: '#404044',
  contrastMediumColor: '#7E7F82',
  contrastHighColor: '#AFB0B3',
  contrastHighColorDarken: '#909195',
  contrastHighColorLighten: '#CECFD1',
  hoverColor: 'rgba(148, 149, 152, .18)',
  hoverColorDarken: '#75767A',
  activeColor: 'rgba(126, 127, 130, 0.20)',
  focusColor: '#1A44EA',
  disabledColor: '#7E7F82',
  errorColor: '#CB3333',
  errorColorDarken: '#9A2727',
  errorSoftColor: '#290A0A',
  errorSoftColorDarken: '#0D0808',
  errorSoftColorLighten: '#331F1F',
  successColor: '#00C77E',
  successColorDarken: '#008A57',
  successSoftColor: '#003320',
  successSoftColorDarken: '#04110C',
  successSoftColorLighten: '#0F432F',
  warningColor: '#DDB84B',
  warningSoftColor: '#2B2208',
  warningSoftColorDarken: '#0D0C07',
  warningSoftColorLighten: '#362F1C',
  infoColor: '#027FFC',
  infoSoftColor: '#001A33',
  infoSoftColorDarken: '#040A11',
  infoSoftColorLighten: '#0F2943'
};

const themes = {
  'light': themeLight,
  'dark': themeDark
};
/* Auto Generated End */

export const getThemedColors = (theme: Theme): ThemedColors => {
  return themes[theme];
};

export const getInvertedThemedColors = (theme: Theme): ThemedColors => {
  return getThemedColors(isThemeDark(theme) ? 'light' : 'dark');
};
