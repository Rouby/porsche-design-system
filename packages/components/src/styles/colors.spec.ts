import { getThemedColors } from './colors';

describe('getThemedColors()', () => {
  it.each<Parameters<typeof getThemedColors>>([['light'], ['dark']])(
    'should return correct colors for theme: %s',
    (theme) => {
      expect(getThemedColors(theme)).toMatchSnapshot();
    }
  );
});
