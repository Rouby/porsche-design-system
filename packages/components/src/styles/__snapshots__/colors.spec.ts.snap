// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getThemedColors() should return correct colors for theme: dark 1`] = `
{
  "activeColor": "rgba(126, 127, 130, 0.20)",
  "backgroundColor": "#0E0E12",
  "backgroundColorDarken": "#000000",
  "backgroundColorLighten": "#292934",
  "backgroundSurfaceColor": "#212225",
  "backgroundSurfaceColorDarken": "#040405",
  "backgroundSurfaceColorLighten": "#3E4045",
  "contrastHighColor": "#AFB0B3",
  "contrastHighColorDarken": "#909195",
  "contrastHighColorLighten": "#CECFD1",
  "contrastLowColor": "#404044",
  "contrastMediumColor": "#7E7F82",
  "disabledColor": "#7E7F82",
  "errorColor": "#CB3333",
  "errorColorDarken": "#9A2727",
  "errorSoftColor": "#290A0A",
  "errorSoftColorDarken": "#0D0808",
  "errorSoftColorLighten": "#331F1F",
  "focusColor": "#1A44EA",
  "hoverColor": "rgba(148, 149, 152, .18)",
  "hoverColorDarken": "#75767A",
  "infoColor": "#027FFC",
  "infoSoftColor": "#001A33",
  "infoSoftColorDarken": "#040A11",
  "infoSoftColorLighten": "#0F2943",
  "primaryColor": "#FBFCFF",
  "primaryColorDarken": "#BECEFF",
  "successColor": "#00C77E",
  "successColorDarken": "#008A57",
  "successSoftColor": "#003320",
  "successSoftColorDarken": "#04110C",
  "successSoftColorLighten": "#0F432F",
  "warningColor": "#DDB84B",
  "warningSoftColor": "#2B2208",
  "warningSoftColorDarken": "#0D0C07",
  "warningSoftColorLighten": "#362F1C",
}
`;

exports[`getThemedColors() should return correct colors for theme: light 1`] = `
{
  "activeColor": "rgba(148, 149, 152, 0.20)",
  "backgroundColor": "#FFF",
  "backgroundColorDarken": "#E0E0E0",
  "backgroundColorLighten": "#FFFFFF",
  "backgroundSurfaceColor": "#EEEFF2",
  "backgroundSurfaceColorDarken": "#CBCED7",
  "backgroundSurfaceColorLighten": "#FFFFFF",
  "contrastHighColor": "#535457",
  "contrastHighColorDarken": "#353638",
  "contrastHighColorLighten": "#717276",
  "contrastLowColor": "#D8D8DB",
  "contrastMediumColor": "#949598",
  "disabledColor": "#949598",
  "errorColor": "#E7323B",
  "errorColorDarken": "#C51720",
  "errorSoftColor": "#FCE8E9",
  "errorSoftColorDarken": "#EED7D9",
  "errorSoftColorLighten": "#FFFFFF",
  "focusColor": "#1A44EA",
  "hoverColor": "rgba(148, 149, 152, .18)",
  "hoverColorDarken": "#75767A",
  "infoColor": "#1E5BEB",
  "infoSoftColor": "#E8EEFD",
  "infoSoftColorDarken": "#D6DEF0",
  "infoSoftColorLighten": "#FFFFFF",
  "primaryColor": "#010205",
  "primaryColorDarken": "#000000",
  "successColor": "#32B85B",
  "successColorDarken": "#258843",
  "successSoftColor": "#EBFAF0",
  "successSoftColorDarken": "#DDEAE1",
  "successSoftColorLighten": "#FFFFFF",
  "warningColor": "#FECC1B",
  "warningSoftColor": "#FFF9E6",
  "warningSoftColorDarken": "#F5ECD2",
  "warningSoftColorLighten": "#FFFFFF",
}
`;
